src/ui/src/builder/BuilderApp.vue

```diff
@@ -136,6 +138,12 @@ import BuilderInstanceTracker from "./BuilderInstanceTracker.vue";
 import BuilderCollaborationTracker from "./BuilderCollaborationTracker.vue";
 import BaseNote from "@/components/core/base/BaseNote.vue";
 
+const isAutogenShown = ref(false);
+provide(injectionKeys.isAutogenShown, isAutogenShown);
+provide(injectionKeys.showAutogen, () => {
+	isAutogenShown.value = true;
+});
```

Comment:
I don't think you need two `provide` here. You could simply have

```suggestion
provide(injectionKeys.isAutogenShown, ref(false));
```

And then open the autogen like this

```ts
const isAutogenShown = inject(injectionKeys.isAutogenShown, ref(false));

function showAutogen() {
  isAutogenShown.value = true;
}
```

Also, I think it could be better to rename to `isAutogenModalShown`